app/controllers/admins/categories_controller.rb

```diff
@@ -0,0 +1,56 @@
+module Admins
+  class CategoriesController < BaseController
+    def index
+      @categories = Category.all
+    end
+
+    def show
+      category
+    end
+
+    def new
+      @category = Category.new
+    end
+
+    def edit
+      category
+    end
+
+    def create
+      @category = Category.new(category_params)
+      if @category.save
+        redirect_to admins_categories_path, notice: "Category #{category.title} was successfully created."
+      else
+        render :new
+      end
+    end
+
+    def update
+      if category.update(category_params)
+        redirect_to admins_category_path(category), notice: "Category #{category.title} was updated"
+      else
+        render :edit
+      end
+    end
+
+    def destroy
+      if category.rewards.any?
```

Comment:
prevent destroying category with rewards

Reply:
Maybe you could use Policy Object for that?